plugin/trino-redis/src/main/java/io/trino/plugin/redis/RedisRecordCursor.java

```diff
@@ -139,63 +148,83 @@ private boolean endOfData()
         return false;
     }
 
-    private boolean nextRow(String keyString)
+    private boolean nextMultipleRows()
     {
-        fetchData(keyString);
+        fetchData();
 
-        byte[] keyData = keyString.getBytes(StandardCharsets.UTF_8);
+        for (int i = 0; i < keys.size(); i++) {
+            String keyString = keys.get(i);
+            byte[] keyData = keyString.getBytes(StandardCharsets.UTF_8);
 
-        byte[] valueData = EMPTY_BYTE_ARRAY;
-        if (valueString != null) {
-            valueData = valueString.getBytes(StandardCharsets.UTF_8);
-        }
+            byte[] stringValueData = EMPTY_BYTE_ARRAY;
+            Map<String, String> hashValueMap = new HashMap<>();
+            switch (split.getValueDataType()) {
```

Comment:
`RedisDataType` is determined before entering this for loop. Please consider restructuring the code.